### PR TITLE
Add key vault common properties configuration supports all key vault features

### PR DESCRIPTION
**Pull request by hui1110**
As title.
Add properties prefixed with `spring.cloud.azure.keyvault`, so that common properties endpoint can be shared. Support all features in one starter.

### CODE REVIEW

File: sdk/spring/CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +18,8 @@ This section includes changes in `spring-cloud-azure-autoconfigure` module.
 #### Breaking Changes
 + Add `AzureStorageConfiguration` to make Azure storage service share common property configuration [#29094](https://github.com/Azure/azure-sdk-for-java/pull/29094).
     +  Add properties `spring.cloud.azure.storage.endpoint`, `spring.cloud.azure.storage.account-key`, `spring.cloud.azure.storage.sas-token`, `spring.cloud.azure.storage.connection-string`, `spring.cloud.azure.storage.account-name`.
++ Add `AzureKeyVaultConfiguration` to make Azure Key Vault service share common property configuration [#29163](https://github.com/Azure/azure-sdk-for-java/pull/29163).
+    + Add properties `spring.cloud.azure.keyvault`.
```
Comment (saragluna) [inline sdk/spring/CHANGELOG.md:22]: They'd better be put in the New Features?

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/common/AzureKeyVaultConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.cloud.autoconfigure.keyvault.common;
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/common/AzureKeyVaultConfiguration.java:4]: why not put it in `com.azure.spring.cloud.autoconfigure.keyvault`, to be consistent with storage?

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/common/AzureKeyVaultConfiguration.java (hunk 2 above)
Comment (hui1110) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/common/AzureKeyVaultConfiguration.java]: updated

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/properties/utils/AzureServicePropertiesUtils.java [hunk 3]
```diff
@@ -3,31 +3,44 @@
...
+
+        if (source.getClient() instanceof ClientOptionsProvider.HttpClientOptions && target.getClient() instanceof ClientOptionsProvider.HttpClientOptions) {
+            BeanUtils.copyProperties(source.getClient(), target.getClient());
+            ClientOptionsProvider.HttpClientOptions targetClient = (ClientOptionsProvider.HttpClientOptions) target.getClient();
+            ClientOptionsProvider.HttpClientOptions sourceClient = (ClientOptionsProvider.HttpClientOptions) source.getClient();
+            targetClient.getHeaders().addAll(sourceClient.getHeaders());
+            BeanUtils.copyProperties(sourceClient.getLogging(), targetClient.getLogging());
+            targetClient.getLogging().getAllowedHeaderNames().addAll(sourceClient.getLogging().getAllowedHeaderNames());
+            targetClient.getLogging().getAllowedQueryParamNames().addAll(sourceClient.getLogging().getAllowedQueryParamNames());
+        }
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/properties/utils/AzureServicePropertiesUtils.java:42]: why do we need this?

File: sdk/spring/spring-cloud-azure-core/src/test/java/com/azure/spring/cloud/core/util/AzurePropertiesUtilsTests.java [hunk 4]
```diff
@@ -409,6 +415,41 @@ void testCopyPropertiesSourceNotChanged() {
...
+        assertEquals(5, target.getClient().getMaximumConnectionPoolSize());
+        assertEquals("global-application-id", target.getClient().getApplicationId());
+        assertEquals(HttpLogDetailLevel.BODY_AND_HEADERS, target.getClient().getLogging().getLevel());
+        Set<String> allowedHeaderNames = new HashSet<>();
+        allowedHeaderNames.add("abc");
+        assertEquals(allowedHeaderNames, target.getClient().getLogging().getAllowedHeaderNames());
+        assertEquals(1, target.getClient().getHeaders().size());
+    }
+
+    static class AzureHttpClientProperties extends AzureHttpSdkProperties {
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-core/src/test/java/com/azure/spring/cloud/core/util/AzurePropertiesUtilsTests.java:450]: Also add one test case for the AMQP client properties

Comment (hui1110) [reply sdk/spring/spring-cloud-azure-core/src/test/java/com/azure/spring/cloud/core/util/AzurePropertiesUtilsTests.java:500]: added

File: sdk/spring/spring-cloud-azure-core/src/test/java/com/azure/spring/cloud/core/util/AzurePropertiesUtilsTests.java [hunk 5]
```diff
@@ -409,6 +415,41 @@ void testCopyPropertiesSourceNotChanged() {
...
+        source.getClient().setResponseTimeout(Duration.ofSeconds(4));
+        source.getClient().setReadTimeout(Duration.ofSeconds(4));
+        source.getClient().setConnectTimeout(Duration.ofSeconds(2));
+        source.getClient().setMaximumConnectionPoolSize(5);
+        source.getClient().setConnectionIdleTimeout(Duration.ofSeconds(5));
+        source.getClient().setApplicationId("global-application-id");
+        source.getClient().getLogging().setLevel(HttpLogDetailLevel.BODY_AND_HEADERS);
+        source.getClient().getLogging().getAllowedHeaderNames().add("abc");
+        source.getClient().getHeaders().add(headerProperties);
+        AzureHttpClientProperties target = new AzureHttpClientProperties();
```
[the remark this answers is not in the document]
Comment (hui1110) [reply sdk/spring/spring-cloud-azure-core/src/test/java/com/azure/spring/cloud/core/util/AzurePropertiesUtilsTests.java:436]: added

File: sdk/spring/spring-cloud-azure-core/src/test/java/com/azure/spring/cloud/core/util/AzurePropertiesUtilsTests.java [hunk 6]
```diff
@@ -409,6 +417,119 @@ void testCopyPropertiesSourceNotChanged() {
...
+        assertEquals("target-global-application-id", target.getClient().getApplicationId());
+        assertEquals(HttpLogDetailLevel.HEADERS, target.getClient().getLogging().getLevel());
+        Set<String> allowedHeaderNames = new HashSet<>();
+        allowedHeaderNames.add("header-name1");
+        allowedHeaderNames.add("header-name2");
+        assertEquals(allowedHeaderNames, target.getClient().getLogging().getAllowedHeaderNames());
+        assertEquals(1, target.getClient().getHeaders().size());
+    }
+
+    static class AzureHttpClientProperties extends AzureHttpSdkProperties {
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-core/src/test/java/com/azure/spring/cloud/core/util/AzurePropertiesUtilsTests.java:500]: We'd better move the static inner class to the end of this class file

**Review by saragluna — APPROVED** [review]

**Pull request merged**